### PR TITLE
[#41] Add support for PUT PATCH and DELETE method in forms

### PR DESCRIPTION
**Pull request by erdlet**

### CODE REVIEW

**Comment by erdlet** [comment]
Just checked the build and I found out, that the failing test for the `DefaultFormEntityProvider` can fail in case it contains the byte representation of `=`. Opened issue #46 because it doesn't belong directly to this issue.

**Comment by Daniel-Dos** [comment]
great @erdlet . 

thanks .

**Comment by erdlet** [comment]
@Daniel-Dos 

you're welcome :)

File: core/src/main/java/org/eclipse/krazo/forms/HiddenMethodFilter.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    return MediaType.APPLICATION_FORM_URLENCODED_TYPE
+        .isCompatible(requestContext.getMediaType());
+  }
+
+  private String getHiddenMethod(final Form form) {
+    String hiddenMethod = null;
+    final List<String> hiddenFieldValues = form.asMap().getOrDefault(HIDDEN_METHOD_NAME, Collections
+        .emptyList());
+
+    if (!hiddenFieldValues.isEmpty()) {
```
Comment (ivargrimstad) [inline core/src/main/java/org/eclipse/krazo/forms/HiddenMethodFilter.java]: Should there be some check for which methods that are allowed (i.e. check for PUT, DELETE, PATCH)?

Comment (erdlet) [reply core/src/main/java/org/eclipse/krazo/forms/HiddenMethodFilter.java]: I didn't add this check intentionally, as non-HTTP verbs should fail automatically by some exception (tried it yesterday got an OperationNotSupportedException). Sure, OPTION and HEAD don't really make sense as form method, but I think that the developer using this feature should be aware of that.

Comment (ivargrimstad) [reply core/src/main/java/org/eclipse/krazo/forms/HiddenMethodFilter.java]: Makes sense. Thanks!

**Comment by Daniel-Dos** [comment]
I think a good option to add something on the website to RI resources : 

https://www.mvc-spec.org/krazo/ 
https://github.com/mvc-spec/www.mvc-spec.org

**Comment by chkal** [comment]
@gtudan I agree that the mvc-spec.org site would be a great place to provide more details about such Krazo-specific features. Maybe we need some kind of basic structure for a Krazo documentation first?

**Comment by chkal** [comment]
I'll merge this now. Thanks @erdlet for implementing this feature! :beers:

**Pull request merged**